### ISSUE TITLE
Mosaic crashes in create_blank_image()

### ISSUE THREAD

**Issue by pllim** [ejeschke/ginga#95]
Hi again. I dragged an image file from your file browser plugin onto the display area to make a mosaic, I get the following traceback:
```
2015-04-02 13:41:40,111 | I | Mosaic.py:360 (drop_cb) | files dropped: ['file:///mypath/image.fits']
2015-04-02 13:41:40,122 | D | Task.py:929 (execute) | now executing task '<ginga.main.Ginga object at 0x7f058f163c10>.task1'
2015-04-02 13:41:40,123 | D | Task.py:369 (execute) | Running error_wrap((<bound method Mosaic.mosaic of <Mosaic.Mosaic object at 0x7f050c279690>>, ['file:///mypath/images.fits']), {'new_mosaic': True})
2015-04-02 13:41:40,134 | D | Control.py:471 (guess_filetype) | MIME type is image/fits
2015-04-02 13:41:40,135 | D | Control.py:497 (load_image) | assuming file type: image/fits'
2015-04-02 13:41:40,135 | I | Control.py:507 (load_image) | Loading image from file:///mypath/image.fits kwdargs={'numhdu': None}
2015-04-02 13:41:40,135 | D | AstroImage.py:99 (load_file) | Loading file 'file:///mypath/image.fits' ...
2015-04-02 13:41:40,136 | D | io_fits.py:129 (load_file) | Loading file '/mypath/image.fits' ...
2015-04-02 13:41:40,239 | D | Control.py:523 (load_image) | Successfully loaded file into image object.
2015-04-02 13:41:40,239 | D | ImageView.py:896 (get_data_xy) | before adjustment, win_x=240 win_y=232
2015-04-02 13:41:40,245 | D | ImageView.py:912 (get_data_xy) | data_x=-128 data_y=128
2015-04-02 13:41:40,246 | D | Bindings.py:1649 (window_motion) | Event map for (None, 'nobtn')
2015-04-02 13:41:40,342 | D | Mosaic.py:227 (prepare_mosaic) | image0 rot=0.000000 cdelt1=0.000009 cdelt2=0.000009
2015-04-02 13:41:40,342 | D | Mosaic.py:234 (prepare_mosaic) | creating blank image to hold mosaic
2015-04-02 13:41:40,343 | D | ImageView.py:753 (get_rgb_object) | times: total=0.0000
2015-04-02 13:41:40,343 | D | ImageViewQt.py:210 (render_image) | redraw pixmap=<PySide.QtGui.QPixmap object at 0x7f054417dea8>
2015-04-02 13:41:40,343 | D | ImageViewQt.py:213 (render_image) | drawing to pixmap
2015-04-02 13:41:40,344 | D | ImageViewQt.py:160 (_render_offscreen) | data shape is 1051x1051x4
2015-04-02 13:41:40,344 | E | Control.py:425 (error_wrap) |   File ".../ginga-2.2.20150303042350-py2.7.egg/ginga/Control.py", line 415, in error_wrap
    return method(*args, **kwdargs)

  File ".../ginga-2.2.20150303042350-py2.7.egg/ginga/misc/plugins/Mosaic.py", line 442, in mosaic
    self.prepare_mosaic(image, fov_deg)

  File ".../ginga-2.2.20150303042350-py2.7.egg/ginga/misc/plugins/Mosaic.py", line 242, in prepare_mosaic
    pfx='mosaic')

  File ".../ginga-2.2.20150303042350-py2.7.egg/ginga/util/dp.py", line 64, in create_blank_image
    data = numpy.zeros((height, width), dtype=numpy.float32)

2015-04-02 13:41:40,345 | D | Task.py:379 (execute) | Function returned None
2015-04-02 13:41:40,345 | D | Task.py:954 (execute) | done executing task '<ginga.main.Ginga object at 0x7f058f163c10>.task1'
```

**Comment by ejeschke** [comment]
@pllim, is it possible to get a copy of a FITS file that induces this behavior?

**Comment by pllim** [comment]
Perhaps, but I will need to "sanitize" the data first while keeping the relevant keywords that you need intact. Let me see what I can do.

**Comment by pllim** [comment]
@ejeschke , you should have received an email (or several) from me with access info to the file.

**Comment by ejeschke** [comment]
Hi @pllim, thanks for sending me the FITS file.  On my initial testing it works.  When you load this image into ginga normally, does it have a functioning wcs?  What version of astropy are you using?

**Comment by ejeschke** [comment]
"normally" == not using the Mosaic plugin

**Comment by pllim** [comment]
I can load it normally, and RA and DEC values display fine. I use astropy v1 that comes with Anaconda. It only crashes when I drag and drop that file in the attempt to create a mosaic.

**Comment by pllim** [comment]
With your latest dev version, it does not crash anymore with the sanitized image I gave you, but it still crashes on the original version. I need to look into this before I get back to you.

**Comment by pllim** [comment]
I am able to reproduce the same errors by dragging HST images from file browser plugin into mosaic plugin. Here is how you can obtain the images I used this time:

1. Go to http://hla.stsci.edu
2. Search for "47 Tuc"
3. Filter the results as follow:
  - Detector: ACS/WFC
  - Spectral_Elt: F814W
4. Select these from the table by clicking on the rows:
  - PropID: 10775, VisitNum: 60
  - PropID: 11677, VisitNum: 01, 15
5. Click "Add selection to cart" (small green button above the listing)
6. Click on "Cart" tab and download the 3 files (either zipped or separately)
7. You should now have 3 `drz.fits` files to test with

**Comment by pllim** [comment]
Okay, I think I *might* have figured out what is really happening here. I got Ginga to print out `(height, width)` just before it crashed, and then used the values below:
```
>>> import numpy as np
>>> a = np.zeros((72000, 72000))
MemoryError
```

**Comment by ejeschke** [comment]
@pllim, this is definitely a very memory intensive plugin, since it has to create a numpy array to hold the entire mosaic.  If it is possible to catch and recover from the MemoryError that would be highly desirable.

Can you try to set the FOV down to something smaller?  Or try it on a machine with more RAM or swap space?

**Comment by pllim** [comment]
Theoretically, it is possible to get available memory and compute memory the array would take before creating an array. If computed memory is more than what is available, display error gracefully in GUI instead of crashing with obscure traceback.

For my data, I am attempting to scale down the images while preserving their WCS. Will keep you updated.

**Comment by ejeschke** [comment]
Hi @pllim, I downloaded the three files you suggested.  If I set the FOV to 0.35 deg in the plugin and drag the 3 files onto the pane the mosaic is completed in 3.5 sec.  This is on a 3 year old Dell desktop pc running Ubuntu 14.04 with 16 GB RAM.  When I check with ps, it looks like ginga is taking about 6.1GB virtual memory, but only 693MB resident memory.

**Comment by pllim** [comment]
I am unable to find documentation on how to properly use the plugin. I managed to scaled down the images, which removed the `MemoryError`. Now, when I drag-and-drop a file from file browser plugin, it appears in mosaic display, but the previous image I dropped there disappears. How do I get all three of them stack on each other?

**Comment by ejeschke** [comment]
Either uncheck the option that says "Drop new" or drag the files together.

**Comment by pllim** [comment]
Ah, okay. Thanks!

**Comment by pllim** [comment]
I still think that `MemoryError` could be handled more gracefully. But if you do not wish to fix it, you may close this issue because my original question is answered. Thank you.

**Comment by ejeschke** [comment]
@pllim, before we close this issue, can you locate (with your print statement if necessary) where the MemoryError is generated so that we can attempt to handle it more gracefully?  My past experience with MemoryErrors is not good--by the time you catch them the interpreter is in a state where it cannot recover--but maybe in this case it may be different.  Is this happening for sure in `create_blank_image()` and not in a routine trying to open a FITS file?

**Comment by ejeschke** [comment]
BTW, I would welcome a PR for an option for Ginga to figure out the FOV based on the input images.  I never got around to implementing that.  I'm guessing it would require an initial pass over the headers and use the WCS to calculate the min and max coordinates.  Finally, these would be converted into a field width in degrees.  If a new file gets dragged in that extends the field the existing mosaic data would need to be copied into a new, larger mosaic.

**Comment by pllim** [comment]
This is addressed in #101
